airbyte-integrations/connectors/source-salesforce/source_salesforce/streams.py

```diff
@@ -683,11 +681,8 @@ def request_params(
         order_by_clause = ""
 
         if self.name not in UNSUPPORTED_FILTERING_STREAMS:
-            last_primary_key = (next_page_token or {}).get("primary_key", "")
-            if last_primary_key:
-                where_conditions.append(f"{self.primary_key} > '{last_primary_key}'")
             order_by_fields = ", ".join([self.cursor_field, self.primary_key] if self.primary_key else [self.cursor_field])
-            order_by_clause = f"ORDER BY {order_by_fields} ASC LIMIT {self.page_size}"
+            order_by_clause = f"ORDER BY {order_by_fields} ASC"
```

Comment:
I don't remember who but I've been told the SalesForce API had a limit and this was the reason we had this `limit` statement. Is that true? If yes, isn't it dangerous to remove this limit here?

Reply:
it should not be dangerous to remove limits. The idea of changes in this PR is to replace `LIMIT` and `pagination` with stream slicing, e.g.:
before:
``` SQL
SELECT a,b,c FROM table WHERE id>2 LIMIT 12000;
```
after
``` SQL
SELECT a,b,c FROM table WHERE cursor<2021-01-30 AND cursor>2021-01-01;
```

Reply:
How are we sure that `cursor<2021-01-30 AND cursor>2021-01-01` will return less than 12000 records?

Reply:
Ok! I think my confusion came from the fact that in the declarative model, we stop looping when the next_token_page is null. In the case of Salesforce, this is not case since we always return a next_token_page if the stream as a primary_key and a name. In that case, we stop the iteration using [this condition](https://github.com/airbytehq/airbyte/blob/a3079cfdc598abfd42e3dbcae617053d77034f3f/airbyte-integrations/connectors/source-salesforce/source_salesforce/streams.py#L539). Size is defined at 15000, not 12000 though but we use it explicitly in the LIMIT statement so it should be fine.